regression/esbmc/github_690_z3_cast_success/test.desc

```diff
@@ -1,4 +1,4 @@
-KNOWNBUG
+CORE
 main.c
---ir --z3
+--ir --z3 -Wno-error=implicit-function-declaration
```

Comment:
This is an unrelated change

Reply:
This is relevant. It passed in llvm13 in this PR, but llvm16 has stricter checks and therefore needs to add this flag.

Reply:
Just curious, shouldn't this have been caught by CI when updating to llvm16?
Why only in this PR?

Reply:
IIRC, we ignore the results if it's marked as a KNOWBUG